src/getinfo.sml

```diff
@@ -73,310 +71,215 @@ fun getSpan (f: item * E.env) =
-        (* TODO source spans of XML and SQL sources are weird and you end *)
-        (*   up with eg: a span from eg 1-5 and another from 2-6, makes no sense? *)
-        (*   That's one of the reasons why we're searching for the two things mentioned above *)
-        val result =
-            U.Decl.foldB
-                { kind = fn (env, (k, span), acc) => add env (Kind (k, span)) span acc,
-                  con = fn (env, (k, span), acc) => add env (Con (k, span)) span acc,
-                  exp = fn (env, (k, span), acc) => add env (Exp (k, span)) span acc,
-                  sgn_item = fn (env, (k, span), acc) => add env (Sgn_item (k, span)) span acc,
-                  sgn = fn (env, (k, span), acc) => add env (Sgn (k, span)) span acc,
-                  str = fn (env, (k, span), acc) => add env (Str (k, span)) span acc,
-                  decl = fn (env, (k, span), acc) => add env (Decl (k, span)) span acc,
-                  bind = fn (env, binder) =>
-                            case binder of
-                                U.Decl.RelK x => E.pushKRel env x
-                              | U.Decl.RelC (x, k) => E.pushCRel env x k
-                              | U.Decl.NamedC (x, n, k, co) => E.pushCNamedAs env x n k co
-                              | U.Decl.RelE (x, c) => E.pushERel env x c
-                              | U.Decl.NamedE (x, c) => #1 (E.pushENamed env x c)
-                              | U.Decl.Str (x, n, sgn) => #1 (E.pushStrNamed env x sgn)
-                              | U.Decl.Sgn (x, n, sgn) => #1 (E.pushSgnNamed env x sgn)
-                }
-                env
-                { smallestgoodpart = NONE
-                , smallest = { item = Str (str, { file = fileName
-                                                , first = { line = 0, char = 0}
-                                                , last = { line = Option.getOpt (Int.maxInt, 99999), char = 0} })
-                             , span = { file = fileName
-                                      , first = { line = 0, char = 0}
-                                      , last = { line = Option.getOpt (Int.maxInt, 99999), char = 0} }
-                             , env = env }
-                }
-                ( L.DStr (Compiler.moduleOf "fileName", 0, (L.SgnError, ErrorMsg.dummySpan), (str, {file = fileName, first = ErrorMsg.dummyPos, last = ErrorMsg.dummyPos}))
-                , {file = fileName, first = ErrorMsg.dummyPos, last = ErrorMsg.dummyPos})
+fun filterSgiItems (items: Elab.sgn_item list) : foundInEnv list =
+    let
+        fun mapF item =
+            case item of
+                (Elab.SgiVal (name, _, c), _) => [FoundExp (name, c)]
+              | (Elab.SgiCon (name, _, k, _), _) => [FoundCon (name, k)]
+              | (Elab.SgiDatatype ds, loc) =>
+                List.concat (List.map (fn (dtx, i, _, cs) =>
+                                          FoundExp (dtx, (Elab.CNamed i, loc))
```

Comment:
Why `FoundExp` and not `FoundCon` for the datatype name here?  Ditto for the next kind of `sgn_item`.

Reply:
Yeah that's probably better naming. I took it from the naming in ElabEnv I think, I don't remember.